gateway/app/schemas/quality_on_demand.py

```diff
@@ -0,0 +1,342 @@
+        str,
+        Field(
+            description="REQUIRED. An access token is a previously acquired token granting access to the target resource.",
+        ),
+    ]
+    accessTokenExpiresUtc: Annotated[
+        datetime,
+        Field(
+            description="REQUIRED. An absolute UTC instant at which the token shall be considered expired.",
+        ),
+    ]
+    accessTokenType: Annotated[
+        AccessTokenType,
+        Field(
+            description="REQUIRED. Type of the access token (See [OAuth 2.0](https://tools.ietf.org/html/rfc6749#section-7.1)).",
+        ),
+    ]
+    refreshToken: Annotated[
+        str,
+        Field(
+            description="REQUIRED. An refresh token credential used to acquire access tokens.",
+        ),
+    ]
+    refreshTokenEndpoint: Annotated[
+        AnyUrl,
+        Field(
+            description="REQUIRED. A URL at which the refresh token can be traded for an access token.",
+        ),
+    ]
+
+
+class ExtendSessionDuration(BaseModel):
+    requestedAdditionalDuration: Annotated[
+        int,
+        Field(
+            ge=1,
+            description="Additional duration in seconds to be added to the current session duration. The overall session duration, including extensions, shall not exceed the maximum duration limit for the QoS Profile.\n",
+            examples=[1800],
+        ),
+    ]
+
+
+class Type(Enum):
+    org_camaraproject_quality_on_demand_v1_qos_status_changed = (
+        "org.camaraproject.quality-on-demand.v1.qos-status-changed"
+    )
+
+
+class Specversion(Enum):
+    field_1_0 = "1.0"
+
+
+class Datacontenttype(Enum):
+    application_json = "application/json"
+
+
+class CloudEvent(BaseModel):
+    id: Annotated[
+        str,
+        Field(
+            description="Identifier of this event, that must be unique in the source context.",
+        ),
+    ]
+    source: Annotated[
+        str,
+        Field(
+            description="Identifies the context in which an event happened in the specific provider implementation.",
+        ),
+    ]
+    type: Annotated[Type, Field(description="The type of the event.")]
+    specversion: Annotated[
+        Specversion,
+        Field(
+            description="Version of the specification to which this event conforms (must be 1.0 if it conforms to Cloudevents 1.0.2 version)",
+        ),
+    ]
+    datacontenttype: Annotated[
+        Optional[Datacontenttype],
+        Field(
+            description='media-type that describes the event payload encoding, must be "application/json" for CAMARA APIs',
+        ),
+    ] = None
+    data: Annotated[
+        Optional[Dict[str, Any]],
+        Field(
+            description="Event notification details payload, which depends on the event type",
+        ),
+    ] = None
+    time: Annotated[
+        datetime,
+        Field(
+            description="Timestamp of when the occurrence happened. It must follow RFC 3339\n",
+        ),
+    ]
```

Comment:
Don't duplicate schemas, and if you look at other APIs you see that the existing CloudEvent is parametrized so that the wrong data isn't set